study.md

```diff
@@ -21,18 +21,20 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+ Callback Hell is confusing.
 ```
 
 Are there any nagging questions relating to functions that you have?
 
 ```md
-<!-- your answer here -->
+Use of callback functions.
```

Comment:
Here's a short script that I wrote to demonstrate callback functions.  Let me know if it's unclear.
```javascript
// This function has two parameters: greeter (a callback function) and name (a
// string).  It will call "greeter" with "name" as its argument.
function greet(greeter, name) {
  console.log(`Executing greeter callback with ${name} as the argument.`);

  greeter(name);

  console.log('Greeting finished.');
}

// A named function to be used as a callback function.
function hello(name) {
  console.log(`Hello, ${name}.`);
}

// Another named function to be used as a callback function.
function hi(name) {
  console.log(`Hi, ${name}.`);
}

// Calling greet() with hello() as its callback function.
greet(hello, 'Fatih');
// Output:
// Executing greeter callback with Fatih as the argument.
// Hello, Fatih.
// Greeting finished.

// Calling greet() with hello() as its callback function.
greet(hi, 'Fatih');
// Output:
// Executing greeter callback with Fatih as the argument.
// Hi, Fatih.
// Greeting finished.

// Calling greet() with an anonymous function as its callback function.
greet(someName => {
  console.log(`Greetings, ${someName}.`);
}, 'Fatih');
// Output:
// Executing greeter callback with Fatih as the argument.
// Greetings, Fatih.
// Greeting finished.
```